src/main/java/com/example/gildedrose/Application.java

```diff
@@ -20,4 +23,16 @@ CommandLineRunner loadDatabase(ItemRepository itemRepository) {
             itemRepository.save(new BackstagePassesItem("BackstagePasses", 5, 25));
         };
     }
+
+    @Bean
+    CommandLineRunner loadValidators(Validators validators) {
```

Comment:
**suggestion** Define your own `Bean` that configures the `Validator` instead of mutating the existing `Validator` in the application context.

This certainly works, but it feels like an unconventional use of the `CommandLineRunner`. What would be more conventional is something like this.

```java
@Bean
Validators validators() {
    QualityValidator qualityValidator = new QualityValidator();
    return new Validators(Arrays.asList(
        new AgedBrieValidator(qualityValidator),
        new NormalValidator(qualityValidator),
        new ConjuredValidator(qualityValidator),
        new BackstagePassesValidator(qualityValidator),
        new SulfurasValidator()
    ))
}
```

The `@Service` annotation on `Validators` tells Spring to instantiate this object and put it in the application context. This makes it available for other classes to autowire an instance of `Validators`, in our case the `Validators` is autowired into the `ItemController`. The thing with autowiring is that it tends proliferates throughout your code.

Instead of using `@Service`, you can use `@Bean` to do the same thing, but the advantage here is that you have full control over how you construct the object.

You would also want to remove the `@Service` annotation from the `Validators` class if you go in this direction.

Reply:
Okay will incorporate